presto-main-base/src/main/java/com/facebook/presto/sql/planner/iterative/rule/MinMaxByToWindowFunction.java

```diff
@@ -120,6 +121,9 @@ else if (!maxByAggregations.isEmpty() && minByAggregations.isEmpty()) {
         else {
             return Result.empty();
         }
+        if (candidateAggregation.values().stream().noneMatch(x -> x.getArguments().get(0).getType() instanceof MapType || x.getArguments().get(0).getType() instanceof ArrayType)) {
```

Comment:
Move the map/array type check here, and apply optimization as long as one of the aggregations are in map/array type